Do some cleanup around terms and wording. (#1)
youennf/codec-agnostic-rtp-payload-format: draft-encrypted-media-rtp-architecture.md

           +----------------------+
                      |  Encrypted Stream:
                      |  - encrypted content
                      |  - metadata of encoded content
                      |  - metadata of raw content
                      V
           +--------------------+
           |   Media Packetizer |
           +--------------------+
                      |  RTP Stream:
                      |  - RTP payloads (from encrypted content)
                      |  - RTP header extensions (from metadata)
                      V
                     ... 

Figure 2: A closer look at impact of a Media Encryptor on the sending side.
```

Media Encryptor
-----------------------

The Media Encoder takes a Raw Stream consisting of raw content and metadata of raw content and generates an Encoded Frame consisting of encoded content (typically an array of bytes), and metadata of encoded content.  The metadata of encoded content describes the encoded content.  {{WebCodecs}} represents, encoded content in units of EncodedAudioChunk and EncodedVideoChunk, and metadata of encoded content in units of EncodedAudioChunkMetadata and EncodedVideoChunkMetadata.

> The Media Encryptor takes an Encode Stream and metadata of raw content and generates an Encrypted Stream consisting of encrypted content.  It also separates out the metadata that will be needed by processing units that do not have encryption keys (and thus can work only with an Encrypted Stream and not with an Encoded Stream).   Metadata needed by such processing units is passed directly to the Media Packetizer, but metadata that should not available by such processing units is not passed directly to the Media Packetizer (but may be included in the encrypted content).  {{SFrame}} is an example of encrypted content.

Reviewer: ```suggestion
The Media Encryptor takes an Encode Stream and metadata of raw content and generates an Encrypted Stream consisting of encrypted content. It also separates out the metadata that will be needed by processing units that do not have encryption keys (and thus can work only with an Encrypted Stream and not with an Encoded Stream).   Metadata needed by such processing units is passed directly to the Media Packetizer, but metadata that should not be available by such processing units is not passed directly to the Media Packetizer (but may be included in the encrypted content).  {{SFrame}} is an example of encrypted content.
```

> It is the responsibility of the application using the Media Encryptor to provide it encoded content in meaningful units of encoded content.

Reviewer: ```suggestion
It is the responsibility of the application using the Media Encryptor to provide its encoded content in meaningful units of encoded content.
```

In the common case of a video codec, the unit of encoded content is a video frame in byte format (h264 annex b for example).

> If a unit of encoded content smaller than a video frame is useful for a particular application, it is the responsibility of the Media Encoder to provideo the Media Encryptor such units.  Each unit of encoded content will be encrypted and sent to the Encrypted Media Packetizer separately, as a unit of encrypted content.  The typical case is a video codec supporting spatial scalability: each spatial layer will typically be split in its own encoded frame.

Reviewer: ```suggestion
If a unit of encoded content smaller than a video frame is useful for a particular application, it is the responsibility of the Media Encoder to provide the Media Encryptor such units. Each unit of encoded content will be encrypted and sent to the Encrypted Media Packetizer separately, as a unit of encrypted content. The typical case is a video codec supporting spatial scalability: each spatial layer will typically be split in its own encoded content unit.
```

                      V
           +----------------------+
           |    Media Decryptor   |
           +----------------------+
                      |  encoded content
                      |  metadata of raw content
                      |  metadata of encoded content
                      V
           +----------------------+
           |   Media Decoder      |
           +----------------------+
                      |  raw content
                      |  metadata of raw content
                      V
                     ... 

Figure 3: A closer look at the impact of a Media Decryptor on the receiving side.
```

On the receiving side, the Encrypted Media Depacketizer reconstructs the encypted content, metadata, and payload type of the encoded content from RTP packets.  
For video content, this can be based on reading the Dependency Descriptor header extension if present.  
The Media Decryptor decrypts the decoded content into encoded content.
The correct Media Decoder is identified by the payload type of the encoded content and the encoded content is passed to the Media Decoder, which produces raw content along with metadata of raw content.

> Intermediaries that do not have encryption keys (such as SFUs) can also reconstruct the encrypted content, metadata, and payload type of the encoded content, but cannot decrypt the enrypted content.  This is particularly useful for servers (such as SFUs) that need to have a basic understanding of each frame they receive so as to decide to forward it or not and to which endpoint.   Such processing units may also repacketize the encrypted content and metadata into other RTP streams.   Such intermediaries must not modify the encrypted content.

Reviewer: ```suggestion
Intermediaries that do not have encryption keys (such as SFUs) can also reconstruct the encrypted content, metadata, and payload type of the encoded content, but cannot decrypt the encrypted content.  This is particularly useful for servers (such as SFUs) that need to have a basic understanding of each frame they receive so as to decide to forward it or not and to which endpoint. Such processing units may also repacketize the encrypted content and metadata into other RTP streams. Such intermediaries must not modify the encrypted content.
```

RTP packet generation and reception
-----------------------

One possible implementation may meet these requirements as follows:
1. The payload type of the encrypted content is put into the RTP header.

> 1. The payload type of the encoded content is first byte of the RTP payload.

Reviewer: ```suggestion
1. The payload type of the encoded content is the first byte of the RTP payload.
```

1. The remaining RTP payload is a chunk of the encoded content broken up such that the concatenation of RTP payloads (after removing the payload type prefix) in order of sequence number is equal to the encoded content.
1. Send SVC encrypted content using one SSRC in ascending layer order, with the same RTP timestamp, and only the last RTP packet of the last spatial layer frame will have the marker bit set to 1.

> 1. Send SVC metadata using the Dependency Descriptor header extension. The first RTP packet of the frame will have its start_of_frame equal to 1 and the last packet will have its end_of_frame equal to 1.  Such information may be procided by WebCodecs SvcOutputMetadata

Reviewer: ```suggestion
1. Send SVC metadata using the Dependency Descriptor header extension. The first RTP packet of the frame will have its start_of_frame equal to 1 and the last packet will have its end_of_frame equal to 1. Such information may be provided to the MediaPacketizer as part of the encoded metadata, for instance by WebCodecs SvcOutputMetadata
```
